dexonline/data_loader.py

```diff
@@ -33,17 +35,149 @@ def unzip(archive_path, folder_choosen):
 
 def download_file(url, local_filename, chunk_size=128):
     response = requests.get(url, stream=True)
-    total_size = int(response.headers.get('content-length', 0))
+    total_size = int(response.headers.get("content-length", 0))
     logging.info("The download with jsons archive will start now!")
-    with tqdm(total=total_size, unit='B', unit_scale=True, unit_divisor=1024, desc=local_filename) as pbar:
-        with open(local_filename, 'wb') as file:
+    with tqdm(
+        total=total_size,
+        unit="B",
+        unit_scale=True,
+        unit_divisor=1024,
+        desc=local_filename,
+    ) as pbar:
+        with open(local_filename, "wb") as file:
             for data in response.iter_content(chunk_size=chunk_size):
                 pbar.update(len(data))
                 file.write(data)
 
     return local_filename
 
 
+class Mapare:
+    def __init__(self, mapping_json) -> None:
+        self.mapping = mapping_json
+    
+    def find_dexonline_pos_id(self, inflectionId):
+        return self.mapping["DEXONLINE_MORPH"].get(
+                str(inflectionId),
+                "UNKNOWN"
+            )[1]
+
+    def find_dexonline_pos_detail(self, inflectionId):
+        return self.mapping["DEXONLINE_MORPH"].get(
+                str(inflectionId),
+                "UNKNOWN"
+            )[0]
+
+class AllInflectedForms:
+    def __init__(self, all_inflected_forms_json):
+        self.all_inflected_forms = all_inflected_forms_json
+
+    def find_all_inflected_forms(self, token_text):
+        return self.all_inflected_forms.get(
+            token_text,
+            [UNIDENTIFIED_TOKEN]
+        )
+
+
+class IdToWordPos:
+    def __init__(self, id_to_word_pos_json):
+        self.id_to_word_pos = id_to_word_pos_json
+
+    def find_id_to_word_pos(self, id):
+        return self.id_to_word_pos.get(
+            str(id),
+            "UNKNOWN"
+        )[0]
+
+    def find_id_to_word_pos_form(self, id):
+        return self.id_to_word_pos.get(
+            str(id),
+            "UNKNOWN"
+        )[0].get("form")
+    
+class IdToInflectedForms:
+    def __init__(self, id_to_inflected_forms_json):
+        self.id_to_inflected_forms = id_to_inflected_forms_json
```

Comment:
cred ca ar fi o abstractizare mai buna daca ai face citirea in interiorul clasei, in loc sa o faci in afara;
spun asta pentru ca estimez ca unele fisiere, ca sa fie mai eficienta incarcarea lor, vor fi stocate in alt format decat json si atunci vei schimba doar functia de load in interiorul clasei